File: backend/src/main/java/com/elbuensabor/entities/Usuario.java
package com.elbuensabor.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "usuario")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Usuario extends Base {
    @Column
    private String usuario;
    @Column
    private String clave;
    @Column
    private String uid;

    @ManyToOne
    @JoinColumn(name = "rol_id")
    private Rol rol;

    @JsonIgnore
    @OneToOne(mappedBy = "usuario")
    private Cliente cliente;

}

File: frontend/el-buen-sabor-frontend/src/app/entidades/Usuario.ts
import { Cliente } from './Cliente';
import { Rol } from './Rol';

export class Usuario {
  id: number = 0;
  usuario: string = '';
  clave: string = '';
  uid: string = '';
  rol: Rol = new Rol();
  cliente!: Cliente ;

  // constructor(cliente: Cliente | null) { // pass the needed instance through the constructor
  //   this.cliente = cliente;
  // }
}